wagtail/admin/widgets/button.py

```diff
@@ -127,48 +141,26 @@ def __init__(
 
 # Base class for all listing buttons
 # This is also used by SnippetListingButton defined in wagtail.snippets.widgets
-class ListingButton(Button):
+class ListingButton(BaseButton):
+    """Top-level button to be displayed in a listing view."""
+
     def __init__(self, label="", url=None, classname="", **kwargs):
         classname = f"{classname} button button-small button-secondary".strip()
         super().__init__(label=label, url=url, classname=classname, **kwargs)
 
 
 class PageListingButton(ListingButton):
-    aria_label_format = None
-    url_name = None
-
-    def __init__(self, *args, page=None, next_url=None, attrs={}, user=None, **kwargs):
-        self.page = page
-        self.user = user
-        self.next_url = next_url
-
-        attrs = attrs.copy()
-        if (
-            self.page
-            and self.aria_label_format is not None
-            and "aria-label" not in attrs
-        ):
-            attrs["aria-label"] = self.aria_label_format % {
-                "title": self.page.get_admin_display_title()
-            }
-        super().__init__(*args, attrs=attrs, **kwargs)
-
-    @cached_property
-    def url(self):
-        if self.page and self.url_name is not None:
-            url = reverse(self.url_name, args=[self.page.id])
-            if self.next_url:
-                url += "?" + urlencode({"next": self.next_url})
-            return url
-
-    @cached_property
-    def page_perms(self):
-        if self.page:
-            return self.page.permissions_for_user(self.user)
+    def __init__(self, *args, **kwargs):
+        warn(
+            "`PageListingButton` is deprecated. "
+            "Use `wagtail.admin.widgets.button.ListingButton` instead.",
+            category=RemovedInWagtail70Warning,
+        )
+        super().__init__(*args, **kwargs)
 
 
-class BaseDropdownMenuButton(Button):
-    template_name = "wagtailadmin/pages/listing/_button_with_dropdown.html"
+class BaseDropdownMenuButton(BaseButton):
+    template_name = "wagtailadmin/shared/button_with_dropdown.html"
```

Comment:
Note that I haven't tested `Media` support for these.

It's simple enough to add something like

```py
    @property
    def media(self):
        media = super().media
        for button in self.dropdown_buttons:
            media += button.media
        return media
```

to `BaseDropdownMenuButton`, _but_ I don't think that will get picked up when we render it as part of a table. While both the `Table` and `Column` do support `Media`, the listing buttons depend on the `instance` that is rendered for the cell's row. The way `Table` picks up the `Column`'s media currently does not take the data into account, it only looks at the columns definition before they are bound to the data:

https://github.com/wagtail/wagtail/blob/01838766692821ffd2a699cf9f5186082b28d1e1/wagtail/admin/ui/tables/__init__.py#L463-L468

I think changing the Table's media logic to use the bound column isn't worth it for now. If people do ask for it, they can at least work around this by adding the media somewhere higher up instead of the buttons.